stacker_blueprints/eks.py

```diff
@@ -0,0 +1,427 @@
+        "\n",
+        "cat /tmp/describe_cluster_result.json | grep certificateAuthorityData | awk '{print $2}' | sed 's/[,\"]//g' | base64 -d >  $CA_CERTIFICATE_FILE_PATH",
+        "\n",
+        "MASTER_ENDPOINT=$(cat /tmp/describe_cluster_result.json | grep endpoint | awk '{print $2}' | sed 's/[,\"]//g')",
+        "\n",
+        "INTERNAL_IP=$(curl -s http://169.254.169.254/latest/meta-data/local-ipv4)", "\n",
+        "sed -i s,MASTER_ENDPOINT,$MASTER_ENDPOINT,g /var/lib/kubelet/kubeconfig", "\n",
+        "sed -i s,CLUSTER_NAME,", cluster_name, ",g /var/lib/kubelet/kubeconfig", "\n",
+        "sed -i s,REGION,", Ref("AWS::Region"), ",g /etc/systemd/system/kubelet.service", "\n",
+        "sed -i s,MAX_PODS,", FindInMap("MaxPodsPerNode", instance_type, "MaxPods"),
+        ",g /etc/systemd/system/kubelet.service", "\n",
+        "sed -i s,MASTER_ENDPOINT,$MASTER_ENDPOINT,g /etc/systemd/system/kubelet.service", "\n",
+        "sed -i s,INTERNAL_IP,$INTERNAL_IP,g /etc/systemd/system/kubelet.service", "\n",
+        "DNS_CLUSTER_IP=10.100.0.10", "\n",
+        "if [[ $INTERNAL_IP == 10.* ]] ; then DNS_CLUSTER_IP=172.20.0.10; fi", "\n",
+        "sed -i s,DNS_CLUSTER_IP,$DNS_CLUSTER_IP,g  /etc/systemd/system/kubelet.service", "\n",
+        "sed -i s,CERTIFICATE_AUTHORITY_FILE,$CA_CERTIFICATE_FILE_PATH,g /var/lib/kubelet/kubeconfig", "\n",
+        "sed -i s,CLIENT_CA_FILE,$CA_CERTIFICATE_FILE_PATH,g  /etc/systemd/system/kubelet.service", "\n",
+        "systemctl daemon-reload", "\n",
+        "systemctl restart kubelet", "\n",
+        "/opt/aws/bin/cfn-signal -e $? ",
+        "         --stack ", Ref("AWS::StackName"),
+        "         --resource NodeGroup ",
+        "         --region ", Ref("AWS::Region"), "\n"
+    ]
+    return Base64(Join("", launch_config_userdata))
+
+
+class Workers(Blueprint):
+    VARIABLES = {
+        "ClusterName": {
+            "type": str,
+            "description": "The name of the cluster for workers to join."
+        },
+        "WorkerSecurityGroupId": {
+            "type": str,
+            "description": "The security group ID which will contain worker "
+                           "nodes."
+        },
+        "MinInstanceCount": {
+            "type": int,
+            "description": "The minimum number of worker nodes for the worker "
+                           "AutoScalingGroup.",
+            "default": 1,
+        },
+        "MaxInstanceCount": {
+            "type": int,
+            "description": "The maximum number of worker nodes for the worker "
+                           "AutoScalingGroup.",
+            "default": 3,
+        },
+        "DesiredInstanceCount": {
+            "type": int,
+            "description": "The desired number of worker nodes for the worker "
+                           "AutoScalingGroup. Defaults to minimum.",
+            "default": -1,
+        },
+        "WorkerSubnets": {
+            "type": str,
```

Comment:
Any reason to not use a `list` type?  It seems like that's what is expected when it's used.

Reply:
I frequently pass a stack output here from another stack, and maybe I did something wrong, but I can't seem to create list outputs from a stack which I can reference here.